lld/MachO/ObjC.cpp

```diff
@@ -478,6 +482,8 @@ class ObjcCategoryMerger {
   std::vector<ConcatInputSection *> &allInputSections;
   // Map of base class Symbol to list of InfoInputCategory's for it
   DenseMap<const Symbol *, std::vector<InfoInputCategory>> categoryMap;
+  // Set for tracking InputSection erased via eraseISec
+  std::unordered_set<InputSection *> erasedIsecs;
```

Comment:
It is usually preferred to use a vector and then sort + uniq elements, instead of using a set.
https://llvm.org/docs/ProgrammersManual.html#a-sorted-vector

Also, can we use `SmallVector` or `SmallPtrSet` here?

Reply:
I see the point in set vs sorted array, but here a hashset is the better approach (`unordered_set`). It has better lookup time than binary search and we need that looking up lots of reloc's.

Reply:
If the data tends to be large, you could use `DenseSet` as opposed to `SmallPtrSet`, both of which are unordered, similar to `unordered_set`. I don't expect much perf distinction in the usage here which temporarily tracks the erased isec like a visited set, so using either one shouldn't matter. But it's more on coding convention or preference on llvm data-structure.